### PR TITLE
Write documentation for lib/suseconnect_register.pm

### PR DESCRIPTION
**Pull request by Zaoliang**
see https://progress.opensuse.org/issues/57038

### CODE REVIEW

File: lib/suseconnect_register.pm [hunk 1]
```diff
@@ -39,6 +51,13 @@ sub suseconnect_registration {
     zypper_call 'lr';
 }
 
+=head2 command_register
+
+ command_register();
```
Comment (asdil12) [inline lib/suseconnect_register.pm]: This does not list the function args.

It should look like this: `command_register($version, $addon, $addon_regcode);`

Also make sure to describe each variable in the description text. And write the vars like this `C<$var>` for nice formatting.

File: lib/suseconnect_register.pm [hunk 2]
```diff
@@ -39,6 +51,13 @@ sub suseconnect_registration {
...
+
+Register sles and it's addon with or without reg code. Precompile regexes, handle zdup migration and resolve potential conflict by zypper for extension or just register a bare system
```
Comment (asdil12) [inline lib/suseconnect_register.pm]: Please make the line shorter by adding a newline.

Comment (Zaoliang) [reply lib/suseconnect_register.pm]: thanks!

File: lib/suseconnect_register.pm [hunk 3]
```diff
@@ -39,6 +51,15 @@ sub suseconnect_registration {
     zypper_call 'lr';
 }
 
+=head2 command_register
+
+ command_register($version, $addon, $addon_regcode);
+
+Register sles and it's extension with or without reg code. The variables used for registion are product version C<$version>, extension name C<$addon> and it's extension registration code C<$addon_regcode>.
```
Comment (asdil12) [inline lib/suseconnect_register.pm]:
```suggestion
Register sles and it's extension with or without reg code.
The variables used for registion are product version C<$version>, extension name C<$addon> and it's extension registration code C<$addon_regcode>.
```

**Pull request merged**